pkgs/tools/system/zcfan/default.nix

```diff
@@ -0,0 +1,40 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+}:
+
+# Testing this requires a Thinkpad or the presense of /proc/acpi/ibm/fan
+stdenv.mkDerivation rec {
+  pname = "zcfan";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "cdown";
+    repo = "zcfan";
+    rev = version;
+    hash = "sha256-XngchR06HP2iExKJVe+XKBDgsv98AEYWOkl1a/Hktgs=";
+  };
+
+  doInstallCheck = true;
+
+  configurePhase = ''
+    substituteInPlace Makefile \
+      --replace "/usr/local" $out
+  '';
+
+  buildPhase = ''
+    make DESTDIR=$out install
+  '';
+
+  installCheckPhase = ''
```

Comment:
Not sure, are here hooks to run for this one ?

Reply:
First time I had tried relying on the inbuilt cmake hook I wasn't able to get an output however after coming back to it it now suddenly wants to work fine. 

Removed the build phase in https://github.com/NixOS/nixpkgs/pull/253579/commits/6409b451bc38bfe3ed3f313308d62b35d35e59d2

Reply:
Uh I think you must be confusing with https://github.com/NixOS/nixpkgs/pull/253561 because there's no cmake here.

I am wondering here if there are hooks like `preInstallCheck` and `postInstallCheck` that should be run with `runHook`. I don't know if there are

Reply:
You're right I was fixing both at the same time. I'll recheck this.

Reply:
Having another look, it doesn't look like there's anything more that could be done in a pre or post install check

Reply:
Ah sorry I've had the idea just now of grepping nixpkgs for `preInstallCheck` and looks like this hook does exist.
```$ grep -r 'runHook preInstallCheck'
os-specific/linux/sgx/samples/default.nix:      runHook preInstallCheck
os-specific/linux/sgx/samples/default.nix:      runHook preInstallCheck
os-specific/linux/sgx/sdk/default.nix:    runHook preInstallCheck
os-specific/linux/procdump/default.nix:    runHook preInstallCheck
applications/science/logic/sharpsat-td/default.nix:    runHook preInstallCheck
applications/science/electronics/verilog/default.nix:    runHook preInstallCheck
applications/networking/instant-messengers/mikutter/default.nix:    runHook preInstallCheck
applications/networking/cluster/tektoncd-cli/default.nix:    runHook preInstallCheck
applications/networking/cluster/nomad-pack/default.nix:    runHook preInstallCheck
applications/networking/cluster/atmos/default.nix:    runHook preInstallCheck
applications/networking/cluster/k3d/default.nix:    runHook preInstallCheck
applications/networking/cluster/terragrunt/default.nix:    runHook preInstallCheck
applications/networking/cluster/ktunnel/default.nix:    runHook preInstallCheck
applications/networking/cluster/nerdctl/default.nix:    runHook preInstallCheck
applications/networking/cluster/driftctl/default.nix:    runHook preInstallCheck
applications/networking/cluster/octant/default.nix:    runHook preInstallCheck
applications/networking/cluster/waypoint/default.nix:    runHook preInstallCheck
applications/networking/cluster/starboard/default.nix:    runHook preInstallCheck
^C
```

And there's also `postInstallCheck` so you can add those 2 hooks to the `installCheckPhase`

Reply:
Oh sorry, I mistook the earlier question as asking if there'd be anything we could do inside a pre or post installCheck phase instead of just adding a `runHook` for them. I'll add `runHook preInstallCheck` and `runHook postInstallCheck`.

Reply:
fixed in https://github.com/NixOS/nixpkgs/pull/253579/commits/96025e995d6f482ec15817c51fbc88cc79c44468